# Exploration via l'interface de TOFLIT18 datascape : quelles données a-t-on sur la région Poitou dans TOFLIT18 ?

## Quelles sources sont pertinentes ?

National best guess : non, se fait au niveau france entière

**Local best guess** : oui couvre product x partner x direction.

Objet général : non, au niveau de la france entière

Résumé : non, au niveau de la france entière

**National toutes direction partenaires manquants** : oui, couvre product x partner x direction pour toute la france + quelques partenaires.

National toutes directions tous partenaires : non, existe uniquement pour 1750

**Local** : oui, couvre product x partner x direction, mais n'existe que de 1714 à 1780

1792 first semester : non, toute la france

National partenaires manquants : non, partenaires manquants

1792-both semester : non, toute la france

Tableau des quantités : non, toute la france

Tableau des quantités : non, toute la france

Tableau des marchandises : non, toute la france

Tableau Général : non, toute la france


## Séries temporelles

Réunion de toutes les sources qui ont des choses pour la direction de La Rochelle.

![](enquetes/toflit-la-rochelle-time-series.png)

[lien vers la visualisation](http://toflit18.medialab.sciences-po.fr/#/exploration/indicators?lines=%5B%7B%22color%22%3A%22%23C14F4C%22%2C%22direction%22%3A%22La_Rochelle%22%2C%22sourceType%22%3A%22Local%22%7D%2C%7B%22color%22%3A%22%23504342%22%2C%22direction%22%3A%22La_Rochelle%22%2C%22sourceType%22%3A%22Local%20best%20guess%22%7D%2C%7B%22color%22%3A%22%23A563B7%22%2C%22direction%22%3A%22La_Rochelle%22%2C%22sourceType%22%3A%22National%20toutes%20directions%20partenaires%20manquants%22%7D%5D)

Notes :

* pour la fin 1780, uniquement des données de la source "National toutes directions partenaire manquant" ?

# Produits

## Approche autour des années 1787-1789

![](enquetes/toflit_products_1787-1789_simplification_national_toutes_directions_partenaires_manquants.png)

[Lien vers la visualisation](http://toflit18.medialab.sciences-po.fr/#/exploration/terms?nodeSize=flows&edgeSize=flows&labelThreshold=3&labelSizeRatio=2&classification=product_simplification&sourceType=National%20toutes%20directions%20partenaires%20manquants&childClassification=product_sitc_simplEN&direction=La_Rochelle&kind=total&dateMin=1787&dateMax=1789)

Notes :

* pour la fin 1780, uniquement des données de la source "National toutes directions partenaire manquant" ?
* croisement avec la normalisation Navigo ?

## Réseau des partenaires

### Approche autour des années 1787-1789

| dates | source | groupement des partenaires |
|---|---|---|
| 1787-1789 | Toutes directions partenaires manquants | Grouping |

![](enquetes/toflit_network_1787-1789_grouping_national_toutes_directions_partenaires_manquants.png)


[Lien vers la visualisation](http://toflit18.medialab.sciences-po.fr/#/exploration/network?classification=partner_grouping&sourceType=National%20toutes%20directions%20partenaires%20manquants&kind=total&dateMin=1787&dateMax=1789&nodeSize=flows&edgeSize=flows&labelThreshold=5&labelSizeRatio=2)

| dates | source | groupement des partenaires |
|---|---|---|
| 1787-1789 | Toutes directions partenaires manquants | Normalisation orthographique |

![](enquetes/toflit_network_1787-1789_orthographic_national_toutes_directions_partenaires_manquants.png)


[Lien vers la visualisation](http://toflit18.medialab.sciences-po.fr/#/exploration/network?classification=partner_orthographic&sourceType=National%20toutes%20directions%20partenaires%20manquants&kind=total&dateMin=1787&dateMax=1789&nodeSize=flows&edgeSize=flows&labelThreshold=5&labelSizeRatio=2)


| dates | source | groupement des partenaires |
|---|---|---|
| 1787-1789 | Toutes directions partenaires manquants | Simplification |

![](enquetes/toflit_network_1787-1789_simplification_national_toutes_directions_partenaires_manquants.png)


[Lien vers la visualisation](http://toflit18.medialab.sciences-po.fr/#/exploration/network?classification=partner_simplification&sourceType=National%20toutes%20directions%20partenaires%20manquants&kind=total&dateMin=1787&dateMax=1789&nodeSize=flows&edgeSize=flows&labelThreshold=5&labelSizeRatio=2)


In [9]:
# bootstrapping du client
from lib.client import Api
import json

client = Api()

client.toflit.get_locations('partner_simplification', {
    "partnerClassification":None,
    "kind":"total",
    "sourceType":"National toutes directions partenaires manquants",
    "dateMin": "1750",
    "dateMax": "1780"
})

[{'partner': 'Angleterre',
  'region': 'Rouen',
  'count': 97,
  'value': 5492227.729034424},
 {'partner': 'Angleterre',
  'region': 'Caen',
  'count': 49,
  'value': 290793.4506378174},
 {'partner': 'Angleterre',
  'region': 'Amiens',
  'count': 117,
  'value': 4357377.050231934},
 {'partner': 'Angleterre',
  'region': 'Bordeaux',
  'count': 42,
  'value': 4428339.0498046875},
 {'partner': 'Angleterre', 'region': 'Narbonne', 'count': 1, 'value': 11300},
 {'partner': 'Angleterre',
  'region': 'Marseille',
  'count': 31,
  'value': 615680.6399307251},
 {'partner': 'Angleterre',
  'region': 'Saint-Quentin',
  'count': 5,
  'value': 62215},
 {'partner': 'Angleterre',
  'region': 'Montpellier',
  'count': 14,
  'value': 1544591.2016143799},
 {'partner': 'Angleterre',
  'region': 'Nantes',
  'count': 45,
  'value': 907683.0400390625},
 {'partner': 'Angleterre', 'region': 'Paris', 'count': 20, 'value': 599963},
 {'partner': 'Angleterre', 'region': 'Châlons', 'count': 2, 'value': 423218},
 {'

In [2]:
def query_larochellepartners(start="1787", end="1789", source="National toutes directions partenaires manquants", classification='partner_simplification'):
    
    flux = client.toflit.get_locations(classification, {
        "partnerClassification":None,
        "kind":"total",
        "sourceType":source,
        "dateMin":start,
        "dateMax":end
    })

    def filter_to_larochelle (f):
        print(f)
        if ('direction' in f and f['direction'] == 'La Rochelle'):
            return True
        else:
            return False
    # print("")
    print ("Recherche de partenaires de la direction de la Rochelle de " + start + " à " + end + " selon la source " + source + "(classification : " + classification + ")")
    # print("")
    if (flux):
        flux = filter(filter_to_larochelle, flux)
        # for f in list(flux):
        #     print( f['partner'] + '(' + str(f['count']) + ' flux)')
        return list(flux)
    else :
        # print("Pas de résultats")
        return None
        
# query_larochellepartners(start="1787", end="1789", source="National toutes directions partenaires manquants", classification='partner_simplification')

sources = client.toflit.get_sources_types()
partners_classifications = ["partner_source","partner_orthographic","partner_simplification","partner_africa","partner_grouping","partner_obrien","partner_sourcename","partner_wars"]
years = ["1787", "1788", "1789"]

results = list()
for year in years:
    for source in sources:
        for partner_classification in partners_classifications:
            matching = query_larochellepartners(start=year, end=year, source=source, classification=partner_classification)
            if (matching is not None):
                print("===")
                print("match")
                print("===")
                results.append({
                    "year":year,
                    "classification":partner_classification,
                    "source":source,
                    "flux":matching
                })


Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_source)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_orthographic)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source 1792-both semester(classification : partner_sourcename)
Recherc

Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Best Guess national customs region(classification : partner_simplification)
{'partner': 'Angleterre', 'region': 'Paris', 'count': 7, 'value': 1006855}
{'partner': 'Angleterre', 'region': 'Montpellier', 'count': 14, 'value': 642000}
{'partner': 'Angleterre', 'region': 'Flandre', 'count': 115, 'value': 18880000}
{'partner': 'Angleterre', 'region': 'Amiens', 'count': 52, 'value': 7024000}
{'partner': 'Angleterre', 'region': 'Marseille', 'count': 33, 'value': 2062000}
{'partner': 'Angleterre', 'region': 'Rouen', 'count': 74, 'value': 11923000}
{'partner': 'Angleterre', 'region': 'Caen', 'count': 22, 'value': 755000}
{'partner': 'Angleterre', 'region': 'Rennes', 'count': 58, 'value': 4146000}
{'partner': 'Angleterre', 'region': 'La Rochelle', 'count': 16, 'value': 1604000}
{'partner': 'Angleterre', 'region': 'Bordeaux', 'count': 49, 'value': 12391000}
{'partner': 'Angleterre', 'region': 'France', 'count':

Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Best Guess national customs region(classification : partner_sourcename)
{'partner': 'Angleterre', 'region': 'Paris', 'count': 7, 'value': 1006855}
{'partner': 'Angleterre', 'region': 'Montpellier', 'count': 14, 'value': 642000}
{'partner': 'Angleterre', 'region': 'Flandre', 'count': 115, 'value': 18880000}
{'partner': 'Angleterre', 'region': 'Amiens', 'count': 52, 'value': 7024000}
{'partner': 'Angleterre', 'region': 'Marseille', 'count': 33, 'value': 2062000}
{'partner': 'Angleterre', 'region': 'Rouen', 'count': 74, 'value': 11923000}
{'partner': 'Angleterre', 'region': 'Caen', 'count': 22, 'value': 755000}
{'partner': 'Angleterre', 'region': 'Rennes', 'count': 58, 'value': 4146000}
{'partner': 'Angleterre', 'region': 'La Rochelle', 'count': 16, 'value': 1604000}
{'partner': 'Angleterre', 'region': 'Bordeaux', 'count': 49, 'value': 12391000}
{'partner': 'Angleterre', 'region': 'France', 'count': 22,

Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_orthographic)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_sourcename)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Compagnie des Indes(classification : partner_wars)
Re

Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source National toutes directions partenaires manquants(classification : partner_simplification)
{'partner': 'Angleterre', 'region': 'Paris', 'count': 7, 'value': 1006855}
{'partner': 'Angleterre', 'region': 'Montpellier', 'count': 14, 'value': 642000}
{'partner': 'Angleterre', 'region': 'Flandre', 'count': 115, 'value': 18880000}
{'partner': 'Angleterre', 'region': 'Amiens', 'count': 52, 'value': 7024000}
{'partner': 'Angleterre', 'region': 'Marseille', 'count': 33, 'value': 2062000}
{'partner': 'Angleterre', 'region': 'Rouen', 'count': 74, 'value': 11923000}
{'partner': 'Angleterre', 'region': 'Caen', 'count': 22, 'value': 755000}
{'partner': 'Angleterre', 'region': 'Rennes', 'count': 58, 'value': 4146000}
{'partner': 'Angleterre', 'region': 'La Rochelle', 'count': 16, 'value': 1604000}
{'partner': 'Angleterre', 'region': 'Bordeaux', 'count': 49, 'value': 12391000}
{'partner': 'Angleterre', 'region': 'Fra

Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source National toutes directions partenaires manquants(classification : partner_sourcename)
{'partner': 'Angleterre', 'region': 'Paris', 'count': 7, 'value': 1006855}
{'partner': 'Angleterre', 'region': 'Montpellier', 'count': 14, 'value': 642000}
{'partner': 'Angleterre', 'region': 'Flandre', 'count': 115, 'value': 18880000}
{'partner': 'Angleterre', 'region': 'Amiens', 'count': 52, 'value': 7024000}
{'partner': 'Angleterre', 'region': 'Marseille', 'count': 33, 'value': 2062000}
{'partner': 'Angleterre', 'region': 'Rouen', 'count': 74, 'value': 11923000}
{'partner': 'Angleterre', 'region': 'Caen', 'count': 22, 'value': 755000}
{'partner': 'Angleterre', 'region': 'Rennes', 'count': 58, 'value': 4146000}
{'partner': 'Angleterre', 'region': 'La Rochelle', 'count': 16, 'value': 1604000}
{'partner': 'Angleterre', 'region': 'Bordeaux', 'count': 49, 'value': 12391000}
{'partner': 'Angleterre', 'region': 'France'

Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_orthographic)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_sourcename)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Résumé(classification : partner_wars)
Recherche de partenaires de la direction de la Rochelle de 1787 à 1787 selon la source Tablea

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Best Guess customs region product x partner(classification : partner_sourcename)
{'partner': 'Outre-mers', 'region': "Colonies Françaises de l'Amérique", 'count': 370, 'value': 37396680}
===
match
===
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Best Guess customs region product x partner(classification : partner_wars)
{'partner': "États-Unis d'Amérique et ses colonies", 'region': "Colonies Françaises de l'Amérique", 'count': 69, 'value': 13372481}
{'partner': 'Espagne et ses colonies', 'region': "Colonies Françaises de l'Amérique", 'count': 76, 'value': 11286983}
{'partner': 'Angleterre et ses colonies', 'region': "Colonies Françaises de l'Amérique", 'count': 65, 'value': 7322188}
{'partner': 'Danemark et ses colonies', 'region': "Colonies Françaises de l'Amérique", 'count': 47, 'value': 564654}
{'partner': 'Suède et ses colonies', 'region': "Colonies França

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Best Guess national customs region(classification : partner_africa)
{'partner': 'Africa', 'region': 'Marseille', 'count': 233, 'value': 1821175}
{'partner': 'Other outre-mers', 'region': 'Marseille', 'count': 1057, 'value': 53989724}
{'partner': 'Other outre-mers', 'region': 'Toulon', 'count': 8, 'value': 5228}
{'partner': 'Other outre-mers', 'region': 'Flandre', 'count': 20, 'value': 5920100}
{'partner': 'Other outre-mers', 'region': 'Rouen', 'count': 53, 'value': 39797317}
{'partner': 'Other outre-mers', 'region': 'Rennes', 'count': 27, 'value': 1802452}
{'partner': 'Other outre-mers', 'region': 'Nantes', 'count': 844, 'value': 50773846}
{'partner': 'Other outre-mers', 'region': 'La Rochelle', 'count': 98, 'value': 5462786}
{'partner': 'Other outre-mers', 'region': 'Bordeaux', 'count': 1432, 'value': 117021199}
{'partner': 'Other outre-mers', 'region': 'Bayonne', 'count': 135, 'value': 1854257}
{'p

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Best Guess national customs region(classification : partner_sourcename)
{'partner': 'Outre-mers', 'region': 'Marseille', 'count': 879, 'value': 9095701}
{'partner': 'Colonies', 'region': 'Marseille', 'count': 410, 'value': 46709534}
{'partner': 'Colonies', 'region': 'Toulon', 'count': 8, 'value': 5228}
{'partner': 'Colonies', 'region': 'Flandre', 'count': 20, 'value': 5920100}
{'partner': 'Colonies', 'region': 'Rouen', 'count': 53, 'value': 39797317}
{'partner': 'Colonies', 'region': 'Rennes', 'count': 27, 'value': 1802452}
{'partner': 'Colonies', 'region': 'Nantes', 'count': 452, 'value': 49847351}
{'partner': 'Colonies', 'region': 'La Rochelle', 'count': 98, 'value': 5462786}
{'partner': 'Colonies', 'region': 'Bordeaux', 'count': 648, 'value': 109461003}
{'partner': 'Colonies', 'region': 'Bayonne', 'count': 85, 'value': 1659347}
{'partner': 'Colonies', 'region': 'Saint-Malo', 'count': 63, 'value': 

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Compagnie des Indes(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Compagnie des Indes(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Compagnie des Indes(classification : partner_sourcename)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Compagnie des Indes(classification : partner_wars)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Local(classification : partner_source)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source Local(classification : partner_orthographic)
{'partner': "Possessions des États-Unis d'Amérique", 'region': "Colonies Françaises de l'Amérique", 'count': 69, 'value': 13372481}
{'partner': 'Possessions espagnoles', 'region'

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions partenaires manquants(classification : partner_simplification)
{'partner': 'Sénégal', 'region': 'Marseille', 'count': 71, 'value': 737841}
{'partner': "Colonies françaises d'Amérique", 'region': 'Marseille', 'count': 29, 'value': 34818819}
{'partner': "Colonies françaises d'Amérique", 'region': 'Toulon', 'count': 8, 'value': 5228}
{'partner': "Colonies françaises d'Amérique", 'region': 'Flandre', 'count': 20, 'value': 5920100}
{'partner': "Colonies françaises d'Amérique", 'region': 'Rouen', 'count': 53, 'value': 39797317}
{'partner': "Colonies françaises d'Amérique", 'region': 'Rennes', 'count': 27, 'value': 1802452}
{'partner': "Colonies françaises d'Amérique", 'region': 'Nantes', 'count': 27, 'value': 44467380}
{'partner': "Colonies françaises d'Amérique", 'region': 'La Rochelle', 'count': 24, 'value': 5290581}
{'partner': "Colonies françaises d'Amérique", 'region': 'Bord

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions partenaires manquants(classification : partner_obrien)
{'partner': 'Sénégal', 'region': 'Marseille', 'count': 71, 'value': 737841}
{'partner': "Colonies françaises d'Amérique", 'region': 'Marseille', 'count': 29, 'value': 34818819}
{'partner': "Colonies françaises d'Amérique", 'region': 'Toulon', 'count': 8, 'value': 5228}
{'partner': "Colonies françaises d'Amérique", 'region': 'Flandre', 'count': 20, 'value': 5920100}
{'partner': "Colonies françaises d'Amérique", 'region': 'Rouen', 'count': 53, 'value': 39797317}
{'partner': "Colonies françaises d'Amérique", 'region': 'Rennes', 'count': 27, 'value': 1802452}
{'partner': "Colonies françaises d'Amérique", 'region': 'Nantes', 'count': 27, 'value': 44467380}
{'partner': "Colonies françaises d'Amérique", 'region': 'La Rochelle', 'count': 24, 'value': 5290581}
{'partner': "Colonies françaises d'Amérique", 'region': 'Bordeaux', '

Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions sans produits(classification : partner_orthographic)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions sans produits(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions sans produits(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions sans produits(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions sans produits(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1788 à 1788 selon la source National toutes directions sans produits(classification : partner_sourcename)
Recherche de p

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source 1792-first semester(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source 1792-first semester(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source 1792-first semester(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source 1792-first semester(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source 1792-first semester(classification : partner_sourcename)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source 1792-first semester(classification : partner_wars)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess customs region product x partner(classification : 

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national customs region(classification : partner_simplification)
{'partner': 'Suède', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Gênes', 'region': 'Montpellier', 'count': 70, 'value': 794163.2819824219}
{'partner': 'Gênes', 'region': 'Marseille', 'count': 248, 'value': 7368619.638168335}
{'partner': 'Gênes', 'region': 'Charleville', 'count': 1, 'value': 5800}
{'partner': 'Espagne', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne', 'region': 'La Rochelle', 'count': 19, 'value': 185433.4552001953}
{'partner': 'Espagne', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'partner': 'Espagne', 'region': 'Bordeaux', 'count': 174, 'value': 3773346.7719802856}
{'partner': 'E

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national customs region(classification : partner_africa)
{'partner': 'Not Outre-mers', 'region': 'Châlons', 'count': 1057, 'value': 8511621.015687943}
{'partner': 'Not Outre-mers', 'region': 'Montpellier', 'count': 1298, 'value': 28160118.051765442}
{'partner': 'Not Outre-mers', 'region': 'Marseille', 'count': 2922, 'value': 201980022.0726471}
{'partner': 'Not Outre-mers', 'region': 'Charleville', 'count': 1073, 'value': 11254768.239967346}
{'partner': 'Not Outre-mers', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Not Outre-mers', 'region': 'Lorient', 'count': 536, 'value': 7863074.656560898}
{'partner': 'Not Outre-mers', 'region': 'La Rochelle', 'count': 839, 'value': 19574871.19916916}
{'partner': 'Not Outre-mers', 'region': 'Grenoble', 'count': 1160, 'value': 33979992.03313446}
{'partner': 'Not Outre-mers', 'region': 'Bordeaux', 'count': 1372, 'value': 131430

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national customs region(classification : partner_obrien)
{'partner': 'Suède et ses colonies', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Gênes', 'region': 'Montpellier', 'count': 70, 'value': 794163.2819824219}
{'partner': 'Gênes', 'region': 'Marseille', 'count': 248, 'value': 7368619.638168335}
{'partner': 'Gênes', 'region': 'Charleville', 'count': 1, 'value': 5800}
{'partner': 'Espagne et ses colonies', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne et ses colonies', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne et ses colonies', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne et ses colonies', 'region': 'La Rochelle', 'count': 19, 'value': 185433.4552001953}
{'partner': 'Espagne et ses colonies', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'partner': 'E

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national customs region(classification : partner_sourcename)
{'partner': 'Suède', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Italie', 'region': 'Montpellier', 'count': 168, 'value': 2103180.0432891846}
{'partner': 'Italie', 'region': 'Marseille', 'count': 1046, 'value': 56564353.03520203}
{'partner': 'Italie', 'region': 'Charleville', 'count': 5, 'value': 50181.600036621094}
{'partner': 'Espagne', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne', 'region': 'La Rochelle', 'count': 19, 'value': 185433.4552001953}
{'partner': 'Espagne', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'partner': 'Espagne', 'region': 'Bordeaux', 'count': 174, 'value': 3773346.7719802856

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national customs region(classification : partner_wars)
{'partner': 'Suède et ses colonies', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Gênes', 'region': 'Montpellier', 'count': 74, 'value': 802224.2819824219}
{'partner': 'Gênes', 'region': 'Marseille', 'count': 248, 'value': 7368619.638168335}
{'partner': 'Gênes', 'region': 'Charleville', 'count': 1, 'value': 5800}
{'partner': 'Espagne et ses colonies', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne et ses colonies', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne et ses colonies', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne et ses colonies', 'region': 'La Rochelle', 'count': 20, 'value': 188110.4552001953}
{'partner': 'Espagne et ses colonies', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'partner': 'Esp

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national partner(classification : partner_orthographic)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national partner(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national partner(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national partner(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national partner(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best Guess national partner(classification : partner_sourcename)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source Best 

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions partenaires manquants(classification : partner_simplification)
{'partner': 'Suède', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Gênes', 'region': 'Montpellier', 'count': 70, 'value': 794163.2819824219}
{'partner': 'Gênes', 'region': 'Marseille', 'count': 248, 'value': 7368619.638168335}
{'partner': 'Gênes', 'region': 'Charleville', 'count': 1, 'value': 5800}
{'partner': 'Espagne', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne', 'region': 'La Rochelle', 'count': 19, 'value': 185433.4552001953}
{'partner': 'Espagne', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'partner': 'Espagne', 'region': 'Bordeaux', 'count': 174, 'value': 3773346.7719802856}


Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions partenaires manquants(classification : partner_africa)
{'partner': 'Not Outre-mers', 'region': 'Châlons', 'count': 1057, 'value': 8511621.015687943}
{'partner': 'Not Outre-mers', 'region': 'Montpellier', 'count': 1298, 'value': 28160118.051765442}
{'partner': 'Not Outre-mers', 'region': 'Marseille', 'count': 2922, 'value': 201980022.0726471}
{'partner': 'Not Outre-mers', 'region': 'Charleville', 'count': 1073, 'value': 11254768.239967346}
{'partner': 'Not Outre-mers', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Not Outre-mers', 'region': 'Lorient', 'count': 536, 'value': 7863074.656560898}
{'partner': 'Not Outre-mers', 'region': 'La Rochelle', 'count': 839, 'value': 19574871.19916916}
{'partner': 'Not Outre-mers', 'region': 'Grenoble', 'count': 1160, 'value': 33979992.03313446}
{'partner': 'Not Outre-mers', 'region': 'Bordeaux', 'count': 1372, '

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions partenaires manquants(classification : partner_obrien)
{'partner': 'Suède et ses colonies', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Gênes', 'region': 'Montpellier', 'count': 70, 'value': 794163.2819824219}
{'partner': 'Gênes', 'region': 'Marseille', 'count': 248, 'value': 7368619.638168335}
{'partner': 'Gênes', 'region': 'Charleville', 'count': 1, 'value': 5800}
{'partner': 'Espagne et ses colonies', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne et ses colonies', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne et ses colonies', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne et ses colonies', 'region': 'La Rochelle', 'count': 19, 'value': 185433.4552001953}
{'partner': 'Espagne et ses colonies', 'region': 'Grenoble', 'count': 1, 'value': 300}


Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions partenaires manquants(classification : partner_sourcename)
{'partner': 'Suède', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Italie', 'region': 'Montpellier', 'count': 168, 'value': 2103180.0432891846}
{'partner': 'Italie', 'region': 'Marseille', 'count': 1046, 'value': 56564353.03520203}
{'partner': 'Italie', 'region': 'Charleville', 'count': 5, 'value': 50181.600036621094}
{'partner': 'Espagne', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne', 'region': 'La Rochelle', 'count': 19, 'value': 185433.4552001953}
{'partner': 'Espagne', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'partner': 'Espagne', 'region': 'Bordeaux', 'count': 174, 'value': 3773

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions partenaires manquants(classification : partner_wars)
{'partner': 'Suède et ses colonies', 'region': 'Châlons', 'count': 5, 'value': 20986}
{'partner': 'Gênes', 'region': 'Montpellier', 'count': 74, 'value': 802224.2819824219}
{'partner': 'Gênes', 'region': 'Marseille', 'count': 248, 'value': 7368619.638168335}
{'partner': 'Gênes', 'region': 'Charleville', 'count': 1, 'value': 5800}
{'partner': 'Espagne et ses colonies', 'region': 'Montpellier', 'count': 160, 'value': 2408402.378479004}
{'partner': 'Espagne et ses colonies', 'region': 'Auch', 'count': 206, 'value': 2369490.401649475}
{'partner': 'Espagne et ses colonies', 'region': 'Lorient', 'count': 21, 'value': 232921.40002441406}
{'partner': 'Espagne et ses colonies', 'region': 'La Rochelle', 'count': 20, 'value': 188110.4552001953}
{'partner': 'Espagne et ses colonies', 'region': 'Grenoble', 'count': 1, 'value': 300}
{'

Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions sans produits(classification : partner_orthographic)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions sans produits(classification : partner_simplification)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions sans produits(classification : partner_africa)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions sans produits(classification : partner_grouping)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions sans produits(classification : partner_obrien)
Recherche de partenaires de la direction de la Rochelle de 1789 à 1789 selon la source National toutes directions sans produits(classification : partner_sourcename)
Recherche de p

In [3]:
# formattage des résultats
results_filtered = filter(lambda r : len(r["flux"]) > 0, results)

print("| année | source | classification | nombre de flux | détail des flux |")
print("|---|---|---|---|---|")
for item in list(results_filtered):
    print('| ' + str(item["year"]) + ' | ' + item["source"] + ' | ' + item["classification"] + ' | ' + str(len(item["flux"])) + '|' + ', '.join(list(map(lambda i : i["partner"] + "(" + str(i["count"]) + ")", item["flux"]))) + ' |')
    # print('| ' + str(item["year"]) + ' | ' + item["source"] + " | " + item["classification"] + len(item["flux"]) ± " | |")    

| année | source | classification | nombre de flux | détail des flux |
|---|---|---|---|---|


### Flux disponibles en provenance de la Rochelle (1787-1789)

| année | source | classification | nombre de partenaires | détail des flux |
|---|---|---|---|---|
| 1787 | National toutes directions partenaires manquants | partner_orthographic | 3|Angleterre(16), Sénégal et Guinée(68), Saint-Domingue(23) |
| 1787 | National toutes directions partenaires manquants | partner_simplification | 3|Angleterre(16), Guinée et Sénégal(68), Saint-Domingue(23) |
| 1787 | National toutes directions partenaires manquants | partner_africa | 3|Not Outre-mers(16), Africa(68), Other outre-mers(23) |
| 1787 | National toutes directions partenaires manquants | partner_grouping | 2|Angleterre(16), Outre-mers(91) |
| 1787 | National toutes directions partenaires manquants | partner_obrien | 3|Angleterre(16), Guinée et Sénégal(68), Saint-Domingue(23) |
| 1787 | National toutes directions partenaires manquants | partner_sourcename | 3|Angleterre(16), Guinée(68), Colonies(23) |
| 1787 | National toutes directions partenaires manquants | partner_wars | 2|Angleterre et ses colonies(16), Colonies françaises(91) |
| 1788 | National toutes directions partenaires manquants | partner_orthographic | 3|Saint-Domingue(74), Colonies françaises de l'Amérique(24), Sénégal et Guinée(1) |
| 1788 | National toutes directions partenaires manquants | partner_simplification | 3|Saint-Domingue(74), Colonies françaises d'Amérique(24), Guinée et Sénégal(1) |
| 1788 | National toutes directions partenaires manquants | partner_africa | 2|Other outre-mers(98), Africa(1) |
| 1788 | National toutes directions partenaires manquants | partner_grouping | 1|Outre-mers(99) |
| 1788 | National toutes directions partenaires manquants | partner_obrien | 3|Saint-Domingue(74), Colonies françaises d'Amérique(24), Guinée et Sénégal(1) |
| 1788 | National toutes directions partenaires manquants | partner_sourcename | 2|Colonies(98), Guinée(1) |
| 1788 | National toutes directions partenaires manquants | partner_wars | 1|Colonies françaises(99) |
| 1789 | National toutes directions partenaires manquants | partner_orthographic | 25|Sénégal(149), Saint-Domingue(45), Étranger(8), Îles françaises de l'Amérique(27), Sénégal et Guinée(1), Petites Îles(356), Bayonne(16), Lorient(22), Espagne(19), Angleterre(84), Dunkerque(33), Portugal(2), Hollande(108), Quatre villes hanséatiques(40), États de l'Empereur(7), Danemark(21), Prusse(19), Amérique(10), États-Unis(2), Saint-Jean de Luz(1), Russie(14), Suède(9), Marseille(60), Allemagne(3), Empereur(5) |
| 1789 | National toutes directions partenaires manquants | partner_simplification | 23|Sénégal(149), Saint-Domingue(45), Monde hors colonies(8), Îles françaises de l'Amérique(27), Guinée et Sénégal(1), Petites Îles(356), Bayonne(16), Lorient(22), Espagne(19), Angleterre(84), Dunkerque(33), Portugal(2), Hollande(108), Villes hanséatiques(40), États de l'Empereur(12), Danemark(21), Prusse(19), États-Unis d'Amérique(12), Saint-Jean de Luz(1), Russie(14), Suède(9), Marseille(60), Allemagne(3) |
| 1789 | National toutes directions partenaires manquants | partner_africa | 3|Africa(150), Other outre-mers(72), Not Outre-mers(839) |
| 1789 | National toutes directions partenaires manquants | partner_grouping | 11|Outre-mers(222), Monde(8), France(488), Espagne(19), Angleterre(84), Portugal(2), Hollande(108), Nord(103), Flandre et autres états de l'Empereur(12), États-Unis d'Amérique(12), Allemagne(3) |
| 1789 | National toutes directions partenaires manquants | partner_obrien | 21|Sénégal(149), Saint-Domingue(45), Îles françaises de l'Amérique(27), Guinée et Sénégal(1), Petites Îles(356), Bayonne(16), Lorient(22), Espagne et ses colonies(19), Angleterre(84), Dunkerque(33), Portugal(2), Hollande(108), Villes hanséatiques(40), États de l'Empereur(12), Danemark(21), Prusse(19), États-Unis d'Amérique(12), Russie(14), Suède et ses colonies(9), Marseille(60), Allemagne(3) |
| 1789 | National toutes directions partenaires manquants | partner_sourcename | 17|Outre-mers(149), Colonies(72), Monde(8), Guinée(1), France(488), Espagne(19), Angleterre(84), Portugal(2), Hollande(108), Villes hanséatiques(40), Flandres(12), Danemark(21), Prusse(19), États-Unis d'Amérique(12), Russie(14), Suède(9), Allemagne(3) |
| 1789 | National toutes directions partenaires manquants | partner_wars | 16|Colonies françaises(195), Divers(8), Îles(383), France(131), Espagne et ses colonies(20), Angleterre et ses colonies(84), Portugal et ses colonies(2), Hollande et ses colonies(108), Villes hanséatiques(40), États de l'Empereur(12), Danemark et ses colonies(21), Prusse(19), États-Unis d'Amérique et ses colonies(12), Nord(14), Suède et ses colonies(9), Allemagne(3) |

# Vers le croisement

## Remarques sur le mapping

Le mapping géographique ne me semble pas trop compliqué : on filtre sur la direction de la Rochelle côté PORTIC ?

C'est le mapping des produits qui peut être plus complexe.

Les méthodes d'agrégation des produits sont dynamiques dans TOFLIT18, statiques dans PORTIC. Besoin d'une classification à la toflit sur les marchandises de PORTIC ?

In [4]:
from vega import VegaLite
import pandas as pd

In [5]:
"""VegaLite({
    "mark": "point",
    "encoding": {
        "x": {
            "type": "quantitative",
            "field": "année"
        },
        "y": {
            "type": "nominal",
            "field": "source"
        }
    }
}, results_filtered)

test = [
    {
        "age": 1,
        "lieu": 5
    },
    {
        "age": 2,
        "lieu": 3
    },
    {
        "age": 5,
        "lieu": 5
    },
    {
        "age": 1,
        "lieu": 1
    },
    {
        "age": 5,
        "lieu": 2
    }
]
VegaLite({
    "mark": "point",
    "encoding": {
        "x": {
            "type": "quantitative",
            "field": "age"
        },
        "y": {
            "type": "quantitative",
            "field": "lieu"
        }
    }
}, pd.DataFrame(test))
"""

'VegaLite({\n    "mark": "point",\n    "encoding": {\n        "x": {\n            "type": "quantitative",\n            "field": "année"\n        },\n        "y": {\n            "type": "nominal",\n            "field": "source"\n        }\n    }\n}, results_filtered)\n\ntest = [\n    {\n        "age": 1,\n        "lieu": 5\n    },\n    {\n        "age": 2,\n        "lieu": 3\n    },\n    {\n        "age": 5,\n        "lieu": 5\n    },\n    {\n        "age": 1,\n        "lieu": 1\n    },\n    {\n        "age": 5,\n        "lieu": 2\n    }\n]\nVegaLite({\n    "mark": "point",\n    "encoding": {\n        "x": {\n            "type": "quantitative",\n            "field": "age"\n        },\n        "y": {\n            "type": "quantitative",\n            "field": "lieu"\n        }\n    }\n}, pd.DataFrame(test))\n'

In [6]:
df = pd.DataFrame(results)
df.head()

year          classification                                       source  \
0  1787    partner_orthographic  Best Guess customs region product x partner   
1  1787  partner_simplification  Best Guess customs region product x partner   
2  1787          partner_africa  Best Guess customs region product x partner   
3  1787        partner_grouping  Best Guess customs region product x partner   
4  1787          partner_obrien  Best Guess customs region product x partner   

  flux  
0   []  
1   []  
2   []  
3   []  
4   []